server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/NumericTermsAggregator.java

```diff
@@ -397,7 +401,7 @@ LongTerms buildResult(long owningBucketOrd, long otherDocCount, LongTerms.Bucket
                 bucketCountThresholds.getShardSize(),
                 showTermDocCountError,
                 otherDocCount,
-                List.of(topBuckets),
+                Arrays.asList(topBuckets),
```

Comment:
Do we need these to be mutable lists?

Reply:
this is to avoid copying the list again. This is how it is done iin most of the implementations and I think it is a bug to do it otherwise.